youtube_dl/extractor/common.py

```diff
@@ -1139,7 +1139,8 @@ def _extract_m3u8_formats(self, m3u8_url, video_id, ext=None,
                 if m3u8_id:
                     format_id.append(m3u8_id)
                 last_media_name = last_media.get('NAME') if last_media and last_media.get('TYPE') != 'SUBTITLES' else None
-                format_id.append(last_media_name if last_media_name else '%d' % (tbr if tbr else len(formats)))
+                if last_media_name or not live:
```

Comment:
Such a feature is already implemented in ```YoutubeDL```: https://github.com/rg3/youtube-dl/blob/bf09af3/youtube_dl/YoutubeDL.py#L1318-L1322